### PR TITLE
refactor sync+async versions of exec

### PR DESCRIPTION
**Pull request by maliroteh-sf**
This PR refactors the sync & async versions of executeCommand and puts them into the common utility class so that they can be shared more easily.

### CODE REVIEW

File: src/common/AndroidUtils.ts [hunk 1]
```diff
@@ -56,13 +56,6 @@ export class AndroidSDKUtils {
         return !AndroidSDKUtils.packageCache.isEmpty();
     }
 
-    public static executeCommand(
```
Comment (maliroteh-sf) [inline src/common/AndroidUtils.ts:59]: There's no point in having this function here when all it does is call into another function 🙂

File: src/common/CommonUtils.ts [hunk 2]
```diff
@@ -21,27 +19,50 @@ export class CommonUtils {
...
+                })
+                .toString();
         } catch (error) {
             CommonUtils.logger.error(`Error executing command '${command}':`);
             CommonUtils.logger.error(`${error}`);
             throw error;
         }
     }
 
+    public static async executeCommandAsync(
```
Comment (maliroteh-sf) [inline src/common/CommonUtils.ts:40]: This is not new code. I just moved it here from `IOSUtils.ts`

File: src/common/IOSEnvironmentSetup.ts [hunk 3]
```diff
@@ -4,22 +4,14 @@
...
 import util from 'util';
 import iOSConfig from '../config/iosconfig.json';
+import { CommonUtils } from './CommonUtils';
 import { IOSUtils } from './IOSUtils';
 import { BaseSetup } from './Requirements';
 
-const exec = util.promisify(childProcess.exec);
-
 export class IOSEnvironmentSetup extends BaseSetup {
-    public static executeCommand(
```
Comment (maliroteh-sf) [inline src/common/IOSEnvironmentSetup.ts:17]: Again, there's no point in having this function here when all it does is call into another function.

File: src/common/IOSUtils.ts [hunk 4]
```diff
@@ -100,27 +102,6 @@ export class IOSUtils {
         });
     }
 
-    public static async executeCommand(
```
Comment (maliroteh-sf) [inline src/common/IOSUtils.ts:103]: Moved this to CommonUtils

**Review by sfdctaka — APPROVED** [review]

**Review by khawkins — APPROVED** [review]
Thanks for cleaning that up! It's certainly a sensible refactor. 👍

**Pull request merged**